### ISSUE TITLE
Add integration test for docker-env and oc-env commands

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#1332]
We had some troubles with docker-env and oc-env commands few months ago. These are quite specific commands as they are bound to different shells on different platforms which can cause problems sometimes. To prevent this it would be nice to create an integration test for these commands.

I did a spike on possibilities for testing these commands. Right now I have an early prototype for Linux which serves as a proof of a concept. In short:

- I spawn a process with bash: `terminal = exec.Command("bash")` and I start it `terminal.Start()`
- I can send a stdin into this process with a pipeline: `inPipe, _ = terminal.StdinPipe()` - `inPipe.Write([]byte(command + "\n"))`
- to be able to check for an exit code on next line I add: `inPipe.Write([]byte("echo exitCode=$?\n"))`
- then I wait for exit code: `exitCode := <- exitCodeChannel`, this channel is from a separate goroutine which checks stdout pipe of "terminal *Cmd"

So one can do:
```gherkin
    Scenario: Executing output from docker-env
       Given user starts terminal instance
        When executing command "minishift $dockerenvstdout"
        When executing command "docker ps"
        When executing command "sleep 10"
```

where dockerenvstdout is stdout from `minishift docker-env`.

TODO:

- [x] implement `When setting scenario variable "dockerenvstdout" to the stdout from executing "minishift docker-env"`, currently it supports only oc commands

- [x] implement shell type recognition - bash, cmd, powershell, macos terminal...

- [x] add support for different shells - the commands for checking the exit code `echo $?`, `echo %errorvalue%` etc

**Comment by LalatenduMohanty** [comment]
@agajdosi Good idea. 

> implement When setting scenario variable "dockerenvstdout" to the stdout from executing "minishift docker-env", currently it supports only oc commands

What do you mean by  `currently it supports only oc commands`?



> implement shell type recognition - bash, cmd, powershell, macos terminal...

I think to start with lets do it for bash. `bash` should also work fine for OS X

**Comment by agajdosi** [comment]
@LalatenduMohanty 
> What do you mean by currently it supports only oc commands?

Currently it is implemented this way:

```go
s.Step(`^setting scenario variable "(.*)" to the stdout from executing "oc (.*)"$`,
		minishift.setVariableExecutingOcCommand)
```
and thus it does not match `setting scenario variable "dockerenvstdout" to the stdout from executing "minishift docker-env"`. It could be just writing 2 or 3 lines, but I need to take a proper look how it works. Did not wanted to spent much time on this as I meant it only as a spike for now.

> implement shell type recognition - bash, cmd, powershell, macos terminal...

Currently I know only about 3 things which will make this more complicated: which command to start, if bash, cmd.exe or powershell.exe; which command to run for exit code `echo $?` vs `echo %errorlevel%`; and which values these check will return - integer vs. boolean. These are really not a hard things, but I am not sure how much more stuff will appear during its implementation. So agree, we probably can start with focusing on bash.

**Comment by agajdosi** [comment]
Work on this issue is stopped, however I will continue on it in upcoming sprints.

**Comment by agajdosi** [comment]
Starting on it again.

**Comment by LalatenduMohanty** [comment]
Resolved via #1453
